### ISSUE TITLE
Upgraded to Microsoft.PowerShell.SDK 6.0.1.1, now can't find System.Management.Automation, Version=6.0.1.0,

### ISSUE THREAD

**Issue by jherby2k** [PowerShell/PowerShell#6270]
Steps to reproduce
------------------

1. Download https://github.com/jherby2k/AudioWorks.git
2. Update AudioWorks.Commands.Tests project's nuget package from Microsoft.PowerShell.SDK 6.0.1 to 6.0.1.1
3. Run tests

Expected behavior
-----------------

Tests pass

Actual behavior
---------------

PowerShell tests all fail with:

> One or more errors occurred. (Could not load file or assembly 'System.Management.Automation, Version=6.0.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.

Environment data
----------------

PSVersion                      6.0.1
PSEdition                      Core
GitCommitId                    v6.0.1
OS                             Microsoft Windows 10.0.16299
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```

**Comment by adityapatwardhan** [comment]
@jherby2k Did you upgrade versions for `Microsoft.PowerShell.Commands.Diagnostics` and `Microsoft.WSMan.Management` to `6.0.1.1` as well?

You can remove those from 6.0.1.1+ since the `Microsoft.PowerShell.SDK` includes them as a dependency.

**Comment by jherby2k** [comment]
I did upgrade all 3, and then noticed the dependencies were added (thanks!), so removed them. Regardless, same issue.

I made sure my build directories were clean, etc.

**Comment by adityapatwardhan** [comment]
@jherby2k After running the tests on upgrading to `6.0.1.1` I am seeing these errors:
```
 System.AggregateException : One or more errors occurred. (Object reference not set to an instance of an object.) (The following constructor parameters did not have matching fixture data: ModuleFixture moduleFixture)
      ---- System.NullReferenceException : Object reference not set to an instance of an object.
      ---- The following constructor parameters did not have matching fixture data: ModuleFixture moduleFixture
      Stack Trace:

        ----- Inner Stack Trace #1 (System.NullReferenceException) -----
        D:\OpenPowershell\aditya\AudioWorks\AudioWorks\AudioWorks.Commands.Tests\ModuleFixture.cs(19,0): at AudioWorks.Commands.Tests.ModuleFixture..ctor()
        ----- Inner Stack Trace #2 (Xunit.Sdk.TestClassException) -----
```

Is this what you see?

**Comment by jherby2k** [comment]
Mmm, sort of. I get the following:

System.AggregateException : One or more errors occurred. (Could not load file or assembly 'System.Management.Automation, Version=6.0.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.) (The following constructor parameters did not have matching fixture data: ModuleFixture moduleFixture)
---- System.IO.FileNotFoundException : Could not load file or assembly 'System.Management.Automation, Version=6.0.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.
---- The following constructor parameters did not have matching fixture data: ModuleFixture moduleFixture

I use an xunit test fixture “Modulefixture” to create the PSCore runspace and load my module, and the constructor is failing because it can’t find System.Management.Automation version 6.0.1.0:

            var state = InitialSessionState.CreateDefault();
            state.ImportPSModule(new[]
            {
                Path.Combine(new DirectoryInfo(Directory.GetCurrentDirectory()).FullName, $"{_moduleProject}.dll")
            });
            Runspace = RunspaceFactory.CreateRunspace(state);
            Runspace.Open();

**Comment by twitchax** [comment]
I am having the same issue.

I think this is due to the fact that `Microsoft.PowerShell.SDK` references `System.Management.Automation` at 6.0.1.1, but the _actual_ version of `System.Management.Automation` in the NuGet package appears to be 6.0.1.0.

**Comment by jherby2k** [comment]
Indeed, if i look in the nuget package 6.0.1.1, all the System.Management.Automation.dll libs are version 6.0.1.0. Surprised this got past QC.

**Comment by twitchax** [comment]
@jherby2k, despite the version issue, you might want to try looking around your dependencies and targets files.  I found a rogue `<PackageReference Include="System.Management.Automation" Version="6.0.0" />` that caused problems.  Fixed that, and everything was good.

**Comment by jherby2k** [comment]
This issue continues to be a problem with 6.0.2, even though System.Management.Automation.dll is now the correct version.

I was able to work around the issue by either copying System.Management.Automation.dll manually to the output directory, or specifying System.Management.Automation.dll or by specifying <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies> in the project directory. I don't understand why this broke as of 6.0.1.1. If I listen for AssemblyLoadContext.Default.Resolving events, I don't see anything fire.

**Comment by jherby2k** [comment]
Found the problem - the project referencing Microsoft.PowerShell.SDK is a test project, which is testing a binary module that uses the PowerShell Standard 5.1 (preview) library. That includes its own stub version of System.Management.Automation. Not sure why this is only a problem as of 6.0.1.1, but whatever. Since its not really a compile-time dependency (i'm importing the module into the runspace), I changed it to an after-build xcopy, and now everthing is fine. In case anyone else encounters this.
